### PR TITLE
move non-public modules to b2sdk._internal

### PR DESCRIPTION
**Pull request by mjurbanski-reef**
to be merge to semver v2.x

### CODE REVIEW

File: b2sdk/_internal/file_version.py [hunk 1]
```diff
@@ -14,14 +14,15 @@
...
+from b2sdk._internal.utils.range_ import Range
+
 from .encryption.setting import EncryptionSetting, EncryptionSettingFactory
 from .file_lock import NO_RETENTION_FILE_SETTING, FileRetentionSetting, LegalHold
 from .http_constants import FILE_INFO_HEADER_PREFIX_LOWER, LARGE_FILE_SHA1, SRC_LAST_MODIFIED_MILLIS
-from .progress import AbstractProgressListener
 from .replication.types import ReplicationStatus
-from .utils import Sha1HexDigest, b2_url_decode
-from .utils.http_date import parse_http_date
-from .utils.range_ import Range
```
Comment (emnoor-reef) [inline b2sdk/_internal/file_version.py:26]: Why are some relative imports converted to absolute imports? In my understanding, the imports here did not need to change.

**Review by emnoor-reef — COMMENTED** [review]
I don't think it is feasible to go through the whole PR for a review.

One thing is very confusing to me. The relative imports in the files inside `_internal` should not have been changed, right? But it looks like some relative imports are changed to absolute imports, others were untouched. I think it is making the diff way bigger than it needs to be, though I am not sure as I did not go though the whole PR.

**Review by emnoor-reef — APPROVED** [review]
Found some references of old paths in the docstrings. Opened a PR against your branch. Check if they are correct, as my changes are mostly mechanical without knowing context.

https://github.com/reef-technologies/b2-sdk-python/pull/362

Other than those, LGTM.

File: b2sdk/version.py [hunk 2]
```diff
@@ -9,15 +9,21 @@
...
+    "USER_AGENT",
+]
 
-PYTHON_VERSION = '.'.join(map(str, sys.version_info[:3]))  # something like: 3.9.1
+VERSION = _version("b2sdk")
 
-USER_AGENT = f'backblaze-b2/{VERSION} python/{PYTHON_VERSION}'
+PYTHON_VERSION = ".".join(map(str, _version_info[:3]))  # something like: 3.9.1
+
+USER_AGENT = f"backblaze-b2/{VERSION} python/{PYTHON_VERSION}"
```
[the remark this answers is not in the document]
Comment (mjurbanski-reef) [reply b2sdk/version.py:29]: `__all__` doesn't prevent someone from `from b2sdk.version import version` , only that `from b2sdk.version import * ` doesn't import anything other than things from `__all__`

now the `from b2sdk.version import _version` is still possible, but clearly `_` is an indication that this is not public

**Comment by mjurbanski-reef** [comment]
merged to v2.x

**Pull request closed without merging**